src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs

```diff
@@ -228,46 +228,44 @@ public override void OnActionExecuted(ActionExecutedContext actionExecutedContex
             {
                 // actionExecutedContext.Result might also indicate a status code that has not yet
                 // been applied to the result; make sure it's also successful.
-                StatusCodeResult statusCodeResult = actionExecutedContext.Result as StatusCodeResult;
-                if (statusCodeResult == null || IsSuccessStatusCode(statusCodeResult.StatusCode))
+                ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
```

Comment:
Are there other return result types other than `ObjectResult` that we should care about? I see that [`IStatusCodeActionResult`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.infrastructure.istatuscodeactionresult?view=aspnetcore-5.0) is the common interface of all action results that have a status code. Why not use that?

Reply:
Looking further, it seems that there are a few other result types that have a `Value`, such as `JsonResult`, and there's also `ContentResult` which has `Content`. Is the assumption here that `ObjectResult` is the only one that has a `Value` that can be processed by `[EnableQuery]`?

Reply:
The interface exists for  net core 2.2 and above. I just went through the code and patched the behaviour as we don't look at other items such as JsonResult and Content result.